refactor: Move mempool RPCs to rpc/mempool

maflcko opened the pull request:
This moves the remaining mempool RPCs to `rpc/mempool`. Previously all mempool RPCs from the `blockchain` category have been moved. This patch moves the ones from the `rawtransactions` category.

In the future, as a follow-up to this refactoring patch, it could be considered whether a new `mempool` category should be introduced.

Beside a clearer code organization, this pull request should also reduce the compile time and space of the `rawtransactions.cpp` file.

Commit:
style: Add static keyword where possible in rpc/mempool

promag reviewed (commented):
Code review ACK fac5a51c47fba21678fb35805e40d00fe7c891a0.

Commit:
Move mempool RPCs to rpc/mempool

Can be reviewed with
--color-moved=dimmed-zebra --color-moved-ws=ignore-all-space

shaavan reviewed (commented):
Concept ACK

- The static keyword is used appropriately in `rpc/mempool.cpp` file. I checked for any other instances where a static keyword could be used and found none.
- The moved code looks good to me. I was able to verify the move-only changes in the second commit successfully.
- I was able to compile the PR successfully on Ubuntu 20.04

MarcoFalke commented on src/rpc/mempool.cpp:
```diff
@@ -463,6 +674,8 @@ void RegisterMempoolRPCCommands(CRPCTable& t)
     static const CRPCCommand commands[]{
         // category     actor (function)
         // --------     ----------------
+        {"rawtransactions", &sendrawtransaction},
+        {"rawtransactions", &testmempoolaccept},
```
I am expecting that everything will move to a new `mempool` category later, so I am leaving as-is for now.

shaavan replied:
I think that's acceptable reasoning.
Other than this, I can't find any blockers for this PR, and hence I think it can be merged.

ACK fac5a51c47fba21678fb35805e40d00fe7c891a0
